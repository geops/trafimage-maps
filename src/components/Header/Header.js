import PropTypes from 'prop-types';
import UIHeader from '@geops/react-ui/components/Header';
import Login from '../Login';
// import { SBBLogo } from '../../img';
import { ReactComponent as SBBLogo } from '../../img/sbb-logo.svg';
import LanguageSelect from '../LanguageSelect';

import './Header.scss';

const propTypes = {
  appBaseUrl: PropTypes.string.isRequired,
};

const Header = ({ appBaseUrl }) => {
  return (
    <UIHeader className="wkp-header">
      <div className="wkp-header-right">
        <Login appBaseUrl={appBaseUrl} />
        <LanguageSelect />
        <div className="wkp-header-sbb-logo-wrapper">
          <SBBLogo focusable={false} className="wkp-header-sbb-logo" />
        </div>
      </div>
    </UIHeader>
  );
};

Header.propTypes = propTypes;

export default Header;
