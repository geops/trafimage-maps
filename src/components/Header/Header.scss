.wkp-header {
  position: absolute;
  right: 0;
  left: 0;
  top: 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background: white;
  box-shadow: 1px 1px 2px rgba(0, 0, 0, 0.4);
  border-bottom: 1px solid #b7b7b7;
  height: 100px;
  z-index: 1;

  .wkp-header-right {
    margin-right: 10px;
  }
}
