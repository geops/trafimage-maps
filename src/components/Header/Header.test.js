import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Header from './Header';

describe('Header', () => {
  test('match snapshots', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Header />);
    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
    // Login & LanguageSelect are snapshoted as 'UNDEFINED' due to React.Memo
    // It will be solved in jest.v25: https://github.com/facebook/jest/issues/9216
  });
});
