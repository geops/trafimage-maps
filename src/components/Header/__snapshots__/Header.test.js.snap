// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header match snapshots 1`] = `
<Header
  className="wkp-header"
  classNameLeft="tm-header-left"
  left={null}
>
  <div
    className="wkp-header-right"
  >
    <Memo(Login)
      appBaseUrl="http://foo.de"
    />
    <Memo(LanguageSelect) />
    <div
      className="wkp-header-sbb-logo-wrapper"
    >
      <ForwardRef(SvgSbbLogo)
        className="wkp-header-sbb-logo"
        focusable={false}
      />
    </div>
  </div>
</Header>
`;
